src/main/java/edu/hawaii/its/api/service/GroupingAssignmentService.java

```diff
@@ -111,7 +111,7 @@ public List<GroupingPath> optInGroupingPaths(String owner, String optInUid) {
         optInPaths.removeAll(includes);
         optInPaths = new ArrayList<>(new HashSet<>(optInPaths));
 
-        return groupingsService.getGroupingPaths(optInPaths);
+        return new GroupingPaths(groupingsService.getGroupingPaths(optInPaths));
```

Comment:
One of Frank's guiding principles when working on the API is to put the functionality as deep as possible. In this case, we should move the functionality to return the `GroupingPaths` class into `groupingsService.getGroupingPaths()`. The groupingsService.getGroupingPaths could look something like this:
```
public List<GroupingPath> getGroupingPaths(List<String> groupPaths) {
    return new GroupingPaths(grouperApiService.findGroupsResults(groupPaths))
}
```
Now you would have add another constructor and setter to `GroupingPaths.java` to that takes in `findGroupsResults`.

Reply:
We decided to leave as it is, because there is call to another service function, isGroupingPath().